crates/ruff_server/src/fix.rs

```diff
@@ -14,11 +15,23 @@ use crate::{
 
 pub(crate) fn fix_all(
     document: &crate::edit::Document,
+    document_url: &lsp_types::Url,
     linter_settings: &LinterSettings,
     encoding: PositionEncoding,
 ) -> crate::Result<Vec<lsp_types::TextEdit>> {
     let source = document.contents();
 
+    let document_path = document_url
+        .to_file_path()
+        .expect("document URL should be a valid file path");
```

Comment:
Can you try what happens if you run `fix_all` on a new, unnamed and unsaved file. Does the new file already have a valid path?

Reply:
It does, although the workspace edit seems to fail in certain cases, for some reason? I'll investigate the issue in a follow-up, since it seems to be unrelated to this change.